src/main/java/com/github/manolo8/darkbot/core/entities/Mine.java

```diff
@@ -1,18 +1,18 @@
 package com.github.manolo8.darkbot.core.entities;
 
 import com.github.manolo8.darkbot.core.itf.Obstacle;
-import com.github.manolo8.darkbot.core.utils.pathfinder.Area;
-import com.github.manolo8.darkbot.core.utils.pathfinder.Circle;
+import com.github.manolo8.darkbot.core.utils.pathfinder.AreaImpl;
+import com.github.manolo8.darkbot.core.utils.pathfinder.CircleImpl;
 
 import static com.github.manolo8.darkbot.Main.API;
 
-public class Mine extends Entity implements Obstacle {
+public class Mine extends Entity implements Obstacle, eu.darkbot.api.entities.Mine {
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Class names shouldn’t shadow simple name of implemented interface (NM_SAME_SIMPLE_NAME_AS_INTERFACE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-class-names-shouldn-t-shadow-simple-name-of-implemented-interface-nm-same-simple-name-as-interface).
This class/interface has a simple name that is identical to that of an implemented/extended interface, except that the interface is in a different package (e.g., alpha.Foo extends beta.Foo). This can be exceptionally confusing, create lots of situations in which you have to look at import statements to resolve references and creates many opportunities to accidentally define methods that do not override methods in their superclasses.